packages/runtime-dom/src/apiCustomElement.ts

```diff
@@ -263,7 +264,7 @@ export class VueElement
         )
       }
       if (_def.shadowRoot !== false) {
-        this.attachShadow({ mode: 'open' })
+        this.attachShadow({ mode: 'open', ..._def.shadowRootOptions })
```

Comment:
Can you add unit tests for it？😎

Reply:
Sorry - https://github.com/jsdom/jsdom/issues/3418. I did create the test, but I've had to disable it because of that missing functionality in jsdom.

But I fixed the problem with using `extend` instead of the spread operator.